passportAuth.go

```diff
@@ -0,0 +1,180 @@
+package gowebdav
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// BasicAuth structure holds our credentials
+type PassportAuth struct {
+	user    string
+	pw      string
+	header  *http.Header
+	cookies []string
+}
+
+// constructor for PassportAuth automatically creates a
+func NewPassportAuth(user, pw, reqUrl string, header *http.Header) (*PassportAuth, error) {
+	p := &PassportAuth{
+		user:   user,
+		pw:     pw,
+		header: header,
+	}
+	err := p.genCookies(reqUrl)
+	if err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+// Authorize the current request
+func (p *PassportAuth) Authorize(c *http.Client, rq *http.Request, path string) error {
+	// prevent redirects to detect subsequent authentication requests
+	rq.Header.Set(XInhibitRedirect, "1")
+	for _, cookie := range p.cookies {
+		cookieParts := strings.Split(cookie, ";")
+		cookieName := strings.Split(cookieParts[0], "=")[0]
+		cookieValue := strings.Split(cookieParts[0], "=")[1]
+
+		rq.AddCookie(&http.Cookie{
+			Name:  cookieName,
+			Value: cookieValue,
+		})
+	}
+	return nil
+}
+
+// Verify verifies if the authentication is good
+func (p *PassportAuth) Verify(c *http.Client, rs *http.Response, path string) (reauth bool, err error) {
+	if rs.StatusCode == 302 {
+		if rs.Header.Get("WWW-Authenticate") != "" {
+			// re-authentication required as we are redirected to the login page
+			reauth = true
+			return
+		} else {
+			// just a redirect, follow it
```

Comment:
do the auth ping-pong in verify. bacause the `rs.Request.Body` is already used.
in this case you must obtain a new body va `rs.Request.GetBody()`.
I would create an instance of the authenticator per request and control the `XInhibitRedirect` based on the flow.
in this case, disable it, and restart it with the help of reauth.

Reply:
> do the auth ping-pong in verify.

Did that. Unsure about (how to understand) the rest.

Reply:
set `p.inhibitRedirect = false` and `redo=true`
in this case your cookies should be already up-to-date